iDiff/utils/docker_utils.go

```diff
@@ -221,9 +204,41 @@ func imageToTarCmd(image string) (string, error) {
 	}
 	imageTarPath := imageName + ".tar"
 	reader := bytes.NewReader(out.Bytes())
-	err = copyToFile(imageTarPath, reader)
+	err := copyToFile(imageTarPath, reader)
 	if err != nil {
 		return "", err
 	}
 	return imageTarPath, nil
 }
+
+func getHistoryList(image string, eng bool) ([]string, error) {
+	validDocker, err := ValidDockerVersion(eng)
+	if err != nil {
+		return []string{}, err
+	}
+	var history []img.HistoryResponseItem
+	if validDocker {
+		ctx := context.Background()
+		cli, err := client.NewEnvClient()
+		if err != nil {
+			return []string{}, err
+		}
+		history, err = cli.ImageHistory(ctx, image)
+		if err != nil {
+			return []string{}, err
+		}
+	} else {
+		glog.Info("Docker version incompatible with api, shelling out to local Docker client.")
+		history, err = getImageHistory(image)
+		if err != nil {
+			return []string{}, err
+		}
+	}
+
+	strhistory := make([]string, len(history))
+	for i, layer := range history {
+		layerDescription := strings.TrimSpace(layer.CreatedBy)
```

Comment:
nit: no need to instantiate a new string each time here. You can either declare the var outside the loop and overwrite each time, or just pass the results of `strings.TrimSpace()` directly to `Sprintf()`.